src/WeightVectors.jl

```diff
@@ -212,15 +215,65 @@ Base.iszero(w::AbstractWeightVector) = w.m[2] == 5
-        rand(rng, UInt64) < m[k2] * (k1 < len) && return Int(signed(m[k2+1]))
+        rand(rng, UInt64) < @inbounds(m[k2]) * (k1 < len) && return _convert(Int, @inbounds(m[k2+1])) # See justification of @inbounds safety, below.
     end
 end
+#=
+Loose justification that @inbounds is safe in sample_within_level
+
+    k2 = 2*(r>>shift)+pos which is bounded by 2*(typemax(UInt64)>>shift)+pos.
+    We check that 2*(typemax(UInt64)>>shift)+pos+1 is in bounds in m so k2 and k2+1 should
+    also be in bounds, as long as they are at least 1, which is also checked.
+
+Formally,
+
+    Claim                                                    | Justificaiton
+    -------------------------------------------------------- | -------------
+    r is a UInt64                                            | From source
+(1) 0 <= r <= 2^64-1                                         | Property of UInt64
+    2 <= shift::Int                                          | From source
+    Shifting a UInt64 right by a positive Int is a monotonically nondecreasing function
+    0>>shift <= r>>shift <= (2^64-1)>>shift                  | Applying a monotonically nondecreasing function on all sides of (1)
+    0 <= r>>shift <= (2^64-1)>>shift                         | Shifting 0 does not change its value
+    (2^64-1)>>shift <= 2^62-1                                | Shifting right by an Int at least 2 divides by 4 (or more), rounding down.
+(2) 0 <= r>>shift <= (2^64-1)>>shift <= 2^62-1               | Chained innequalities
+    Multiplying a UInt64 below 2^63 by two will not overflow.
+    Multiplying a UInt64 by 2 is, in the absence of overflow, monotonically nondecreasing
+    2*0 <= 2*(r>>shift) <= 2*((2^64-1)>>shift) <= 2*(2^62-1) | Applying a monotonically nondecreasing function on all sides of (2)
+(3) 0 <= 2*(r>>shift) <= 2*(2^64>>shift) <= 2^63-2           | Partial evaluation of expressions
+(4) pos::UInt64 <= 2^63                                      | From source
+    Adding A UInt64 that is less than or equal to 2^63 to a UInt that is less than 2^63 will not overflow.
+    Adding a UInt64 to a UInt64 is, in the absence of overflow, monotonically nondecreasing.
+(5) pos <= 2*(r>>shift)+pos <= 2*(2^64>>shift)+pos <= 2^63-2+pos  | Applying a monotonically nondecreasing function on all sides of (3)
+    2^63-2 + pos <= 2^63-2 + 2^63                            | Applying a monotonically nondecreasing function on all sides of (4)
+(6) 2^63-2 + pos <= 2^64-2                                   | Partial evaluation
+(7) pos <= 2*(r>>shift)+pos <= 2*(2^64>>shift)+pos <= 2^64-2 | Transitivity applied to (5) and (6)
+(8) k2 = 2*(r>>shift)+pos                                    | From source
+(9) pos <= k2 <= 2*(2^64>>shift)+pos <= 2^64-2               | Substitution of (8) into (7)
+(10)Incrementing a UInt64 that is less than or equal to 2^64-2 is a monotonically nondecreasing function
```

Comment:
```suggestion
(10) Adding 1 to a UInt64 that is less than or equal to 2^64-2 is a monotonically nondecreasing function
```

Reply:
I had no space here to keep alignment, but I'm fine your way, too.